component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java

```diff
@@ -255,11 +258,11 @@ public Response validateRequestToJoinSpace(@PathParam("spaceId") String spaceId,
     checkAuthenticatedRequest();
 
     Space space = getSpaceService().getSpaceById(spaceId);
-    
-    if (space == null) {
+    OrganizationService organizationService = (OrganizationService) getPortalContainer().getComponentInstanceOfType(OrganizationService.class);
+    User user = organizationService.getUserHandler().findUserByName(userId);
```

Comment:
You could use identityManager as it performs better that OrganizationService org.exoplatform.social.core.manager.IdentityManagerImpl#getIdentity(java.lang.String, java.lang.String, boolean)

Reply:
done